online/src/main/scala/ai/chronon/online/ExternalSourceRegistry.scala

```diff
@@ -90,4 +147,29 @@ class ExternalSourceRegistry extends Serializable {
           )))
     }
   }
+
+  /**
+    * Resolve handler by name, attempting dynamic registration if not found.
+    *
+    * @param name Handler name to resolve
+    * @param externalParts Optional external parts for dynamic registration
+    * @return Option containing the handler if found or successfully created
+    */
+  private def resolveHandler(name: String, externalParts: Option[Seq[ExternalPart]]): Option[ExternalSourceHandler] = {
+    handlerMap.get(name) match {
+      case Some(handler) => Some(handler)
+      case None          =>
+        // Try dynamic registration if external parts are provided
+        externalParts match {
+          case Some(_) =>
+            try {
+              addHandlersByFactory(externalParts)
+              handlerMap.get(name) // Check again after registration attempt
+            } catch {
+              case _: Exception => None
+            }
+          case None => None
+        }
+    }
```

Comment:
maybe we can simplify the code this way
```suggestion
    handlerMap.get(name).orElse {
      externalParts.flatMap { _ =>
        try {
          addHandlersByFactory(externalParts)
          handlerMap.get(name) // Check again after registration attempt
        } catch {
          case _: Exception => None
        }
      }
    }
```